internal/cmd/commands/credentialstorescmd/vault_funcs.go

```diff
@@ -0,0 +1,186 @@
+package credentialstorescmd
+
+import (
+	"github.com/hashicorp/boundary/api/credentialstores"
+	"github.com/hashicorp/boundary/internal/cmd/base"
+)
+
+func init() {
+	extraVaultFlagsFunc = extraVaultFlagsFuncImpl
+	extraVaultActionsFlagsMapFunc = extraVaultActionsFlagsMapFuncImpl
+	extraVaultFlagsHandlingFunc = extraVaultFlagHandlingFuncImpl
+}
+
+const (
+	addressFlagName              = "vault-address"
+	namespaceFlagName            = "vault-namespace"
+	vaultCaCertFlagName          = "vault-ca-cert"
+	tlsServerNameFlagName        = "vault-tls-server-name"
+	tlsSkipVerifyFlagName        = "vault-tls-skip-verify"
+	vaultTokenFlagName           = "vault-token"
+	clientCertificateFlagName    = "vault-client-certificate"
+	clientCertificateKeyFlagName = "vault-client-certificate-key"
+)
+
+type extraVaultCmdVars struct {
+	flagAddress       string
+	flagNamespace     string
+	flagCaCert        string
+	flagVaultToken    string
+	flagClientCert    string
+	flagClientCertKey string
+	flagTlsServerName string
+	flagTlsSkipVerify bool
+}
+
+func extraVaultActionsFlagsMapFuncImpl() map[string][]string {
+	flags := map[string][]string{
+		"create": {
+			addressFlagName,
+			namespaceFlagName,
+			vaultCaCertFlagName,
+			tlsServerNameFlagName,
+			tlsSkipVerifyFlagName,
+			vaultTokenFlagName,
+			clientCertificateFlagName,
+			clientCertificateKeyFlagName,
+		},
+	}
+	flags["update"] = flags["create"]
+	return flags
+}
+
+func extraVaultFlagsFuncImpl(c *VaultCommand, set *base.FlagSets, _ *base.FlagSet) {
+	f := set.NewFlagSet("Vault Credential Store Options")
+
+	for _, name := range flagsVaultMap[c.Func] {
+		switch name {
+		case addressFlagName:
+			f.StringVar(&base.StringVar{
+				Name:   addressFlagName,
+				Target: &c.flagAddress,
+				Usage:  "The vault address.",
+			})
+		case namespaceFlagName:
+			f.StringVar(&base.StringVar{
+				Name:   namespaceFlagName,
+				Target: &c.flagNamespace,
+				Usage:  "The vault namespace the store should use.",
+			})
+		case vaultCaCertFlagName:
+			f.StringVar(&base.StringVar{
+				Name:   vaultCaCertFlagName,
+				Target: &c.flagCaCert,
+				Usage:  "The CA Cert to use when connecting with vault.",
+			})
+		case tlsServerNameFlagName:
+			f.StringVar(&base.StringVar{
+				Name:   tlsServerNameFlagName,
+				Target: &c.flagTlsServerName,
+				Usage:  `The TLS server name used when connecting to vault.`,
```

Comment:
```suggestion
				Usage:  `Name to use as the SNI host when connecting via TLS.`,
```
This is [the wording](https://www.vaultproject.io/docs/agent#tls_server_name) from the Vault agent configuration page.